lib/active_model/serializer/adapter.rb

```diff
@@ -41,8 +41,12 @@ def adapters
         # @param [Class] klass - adapter class itself
         # @example
         #     AMS::Adapter.register(:my_adapter, MyAdapter)
+        # @note The registered name strips out 'ActiveModel::Serializer::Adapter::'
+        #   so that registering 'ActiveModel::Serializer::Adapter::Json' and
+        #   'Json' will both register as 'json'.
         def register(name, klass)
-          adapter_map.update(name.to_s.underscore => klass)
+          name = name.to_s.remove('ActiveModel::Serializer::Adapter::'.freeze)
```

Comment:
I don't think it's worth a constant, but yes, in later Rubies frozen strings reduce object allocations